torch/distributed/_spmd/parallel_mode.py

```diff
@@ -50,6 +54,75 @@ def transform_and_compile(self, gm: GraphModule) -> GraphModule:
         raise NotImplementedError()
 
 
+class DataParallel(ParallelMode):
+    """Data Parallelism mode."""
+
+    def __init__(
+        self,
+        parallel_style: str = "replicate",
```

Comment:
Why is `parallel_style` a `str` instead of enum?

Reply:
`DataParallelStyle` is actually a Enum defined inside `data_parallel.py`, and we used it here inside the constructor too. The reason I choose to use `str` in the constructor is that I don't quite like a `Enum` to be a public facing API as that requires an additional import statement imposed to users, especially when it's just an option with 2-3 choices. 

We should probably study the UX later to decide which is better before the release, this is right now just my preference.